apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/MetricsConfiguration.java

```diff
@@ -50,11 +50,24 @@ public class MetricsConfiguration extends ConfigurationOptionProvider {
         .dynamic(false)
         .buildWithDefault(1000);
 
+    private final ConfigurationOption<Boolean> reporterHealthMetricsEnabled = ConfigurationOption.booleanOption()
+        .key("agent_reporter_health_metrics")
+        .configurationCategory(METRICS_CATEGORY)
+        .description("Enables metrics which capture the health state of the event reporting mechanism.\n" +
+            "Disabled by default because this is a tech preview feature.")
+        .tags("added[not officially added yet]", "internal")
+        .dynamic(false)
```

Comment:
We do periodic reporting, so it seems like it wouldn't be any overhead to check the config each time - which means this could be dynamic

Reply:
Two reasons why chose to not make it dynamic:
 * `disable_metrics`, which is used to disable individual metrics is not dynamic. This makes this config feel less useful / inconsistent to be dynamic.
 * The config option will be removed after the tech preview, because then the metrics will be opt-out. Therefore I didn't bother with the effort of allowing the metric enabled state to change at runtime